addons/survey/static/src/js/survey_form.js

```diff
@@ -0,0 +1,272 @@
+        if ($select.val() === $other.val()) {
+            $comment_input.show();
+        } else {
+            $comment_input.val('');
+            $comment_input.hide();
+        }
+    },
+
+    // TOOLS
+    // -------------------------------------------------------------------------
+    /**
+    * Convert date value in client current timezone (if review mode)
+    */
+    _formatDateValue: function ($date_group) {
+        var input = $date_group.find('input');
+        var date_value = input.val();
+        if (date_value != '') {
+            var moment_date = field_utils.parse.date(date_value);
+            if ($date_group.data('questiontype') === 'datetime') {
+                date_value = field_utils.format.datetime(moment_date, null, {timezone: true});
+            } else {
+                date_value = field_utils.format.date(moment_date, null, {timezone: true});
+            }
+        }
+        input.val(date_value);
+        return date_value
+    },
+
+    /**
+    * Initialize datetimepicker in correct format and with constraints
+    */
+    _initDateTimePicker: function ($date_group) {
+        var disabledDates = []
+
+        var minDate = $date_group.data('mindate') ? field_utils.format.datetime(moment($date_group.data('mindate')), null, {timezone: true}) : moment({ y: 1900 });
+        var maxDate = $date_group.data('maxdate') ? field_utils.format.datetime(moment($date_group.data('maxdate')), null, {timezone: true}) : moment().add(200, "y");
+
+        var datetimepickerFormat = time.getLangDateFormat()
+        if ($date_group.data('questiontype') === 'datetime') {
+            datetimepickerFormat = time.getLangDatetimeFormat()
+        } else {
+            // Include min and max date in selectable values
+            maxDate = moment(maxDate).add(1, "d");
+            minDate = moment(minDate).subtract(1, "d");
+            disabledDates = [minDate, maxDate]
+        }
+
+        $date_group.datetimepicker({
+            format : datetimepickerFormat,
+            minDate: minDate,
+            maxDate: maxDate,
+            disabledDates: disabledDates,
+            useCurrent: false,
+            viewDate: moment(new Date()).hours(0).minutes(0).seconds(0).milliseconds(0),
+            calendarWeeks: true,
+            icons: {
+                time: 'fa fa-clock-o',
+                date: 'fa fa-calendar',
+                next: 'fa fa-chevron-right',
+                previous: 'fa fa-chevron-left',
+                up: 'fa fa-chevron-up',
+                down: 'fa fa-chevron-down',
+            },
+            locale : moment.locale(),
+            allowInputToggle: true,
+            keyBinds: null,
+        });
+    },
+
+    /**
+    * Clean all form errors in order to clean DOM before a new validation
+    */
+    _resetErrors: function () {
+        $('.js_errzone').html('').hide();
+        $('.o_survey_error').html('').hide();
+    },
+
+});
```

Comment:
Don't forget to `return publicWidget.registry.SurveyFormWidget;` to make the widget extendable.